### ISSUE TITLE
Number as option alias not working as advertised

### ISSUE THREAD

**Issue by madflow** [yargs/yargs#1454]
Hello!

When using a number as alias - it is advertised in the `help` section as `-1` - but this does not work. (The option stays undefined).

```
// run.js
const argv = require('yargs')
  .option('negativeone', {
    alias: '1',
    type: 'boolean',
    description: 'Negative one'
  }).argv;

console.log(argv);
```
```
➜  yargs-test node run.js --help
Optionen:
  --help             Hilfe anzeigen                                    [boolean]
  --version          Version anzeigen                                  [boolean]
  --negativeone, -1  Negative one                                      [boolean]
```

> ➜  yargs-test node ./run.js --negativeone
{ '1': true, _: [], negativeone: true, '$0': 'run.js' }

> ➜ yargs-test node ./run.js -1           
{ _: [ -1 ], '$0': 'run.js' }

> ➜ yargs-test node ./run.js --1
{ '1': true, _: [], negativeone: true, '$0': 'run.js' }

**Comment by mleguen** [comment]
@madflow Did you try setting the `parse-numbers` option to false?

```js
yargs.parserConfiguration({
  "parse-numbers": false
})
```

**Comment by mleguen** [comment]
No, it does not work either.

**Comment by bcoe** [comment]
do we want to support numbers as options, is a question worth asking ... this feels a little weird.

**Comment by madflow** [comment]
> do we want to support numbers as options, is a question worth asking ... this feels a little weird.

@bcoe I am not sure at whom this question is directed - but since I logged this issue: It is weird and felt nonintuitive in our "downstream" application. We removed the number alias.

**Comment by mleguen** [comment]
@bcoe I also fell its weird (but maybe some people use it?)

However, if we decide we do support numbers as options, shouldn'd we **explicitely** not support it (ie throw when a number is passed as a key or an alias, and stop parsing `--1` as `{ '1': true }` in yargs-parser)?

**Comment by benjie** [comment]
I don't think numeric CLI flags are weird; quite a few of the standard unixy tools use numeric CLI flags. I'm sure many of us have used `kill -9` from time to time, but here's a few others off the top of my head:

#### grep

```
$ grep --help
Usage: grep [OPTION]... PATTERN [FILE]...
Search for PATTERN in each FILE.
[...]
  -C, --context=NUM         print NUM lines of output context
  -NUM                      same as --context=NUM
```

#### gzip

```
$ gzip --help
Usage: gzip [OPTION]... [FILE]...
Compress or uncompress FILEs (by default, compress FILES in-place).
[...]
  -V, --version     display version number
  -1, --fast        compress faster
  -9, --best        compress better
```

#### ssh

```
$ ssh 
usage: ssh [-46AaCfGgKkMNnqsTtVvXxYy] [...]
```

#### rsync

```
$ rsync --help
rsync  version 3.1.2  protocol version 31
Copyright (C) 1996-2015 by Andrew Tridgell, Wayne Davison, and others.
[...]
     --noatime               do not alter atime when opening source files
 -4, --ipv4                  prefer IPv4
 -6, --ipv6                  prefer IPv6
```

#### xargs

```
$ xargs --help
Usage: xargs [OPTION]... COMMAND [INITIAL-ARGS]...
Run COMMAND with arguments INITIAL-ARGS and more arguments read from input.

Mandatory and optional arguments to long options are also
mandatory or optional for the corresponding short option.
  -0, --null                   items are separated by a null, not whitespace;
                                 disables quote and backslash processing and
                                 logical EOF processing
```

**Comment by bcoe** [comment]
@benjie see https://github.com/yargs/yargs/pull/1493, we thought of a couple other examples as well.

**Comment by bcoe** [comment]
Is it enough that we document the option as `--1` (CC: @mleguen); I don't think we necessarily want to extend the underlying parser to treat `[0 - 9]` as flag arguments, I see the argument for:

```
--9
--100
```

**Comment by benjie** [comment]
Is there a precedent for `--9` etc options in other tools?

**Comment by bcoe** [comment]
@benjie the tool I had in mind was `node-tap`, which has a `--100`.

**Comment by mleguen** [comment]
Closed by #1493
